### ISSUE TITLE
[AI Enhancement] Let AI players launch a large charge

### ISSUE THREAD

**Issue by NetsuNegi** [Phobos-developers/Phobos#526]
### Description:
AI players often put a large number of idle units on the base, which is easy to cause traffic jams and then lead to the Caton phenomenon. It's time to strengthen AI players so that they can attack regularly with all combat units according to the set time!

### Possible Implementation:
rulesmd.ini
[General]
AIAllToAttackTime=int,int,int ( easy, medium, hard. If set 0 that means AI players in this difficulty will cannot launch this charge)

**Comment by Zorbung** [comment]
This is the same as #413 but yeah, it would be nice to have it. It's the one reason I play the game with "short game" off, to get that huge wave at the end when AI sells everything and goes on hunt.

**Comment by Zorbung** [comment]
@NetsuNegi If this has been really completed, can you please document it? I can't find any updates on this issue, thanks.

**Comment by NetsuNegi** [comment]
> @NetsuNegi If this has been really completed, can you please document it? I can't find any updates on this issue, thanks.

No, it just that I think no one will do this, so I close it. I will try it if possible.

**Comment by NetsuNegi** [comment]
> @NetsuNegi If this has been really completed, can you please document it? I can't find any updates on this issue, thanks.

It is not difficult to implement, but there are some problems. Such as harvesters will also join the charge.

**Comment by NetsuNegi** [comment]
If people can accept that the harvester also joins the charge, I will submit a new PR.

**Comment by FS-21** [comment]
Why not iterate through all units and send the hunt signal? Probably the units should leave the AI Team's Taskforce before and then send the hunt mission. In that way you can check specific units.

If that would be a new script type action should affect by default only the owner's team and make "ALL AI players" optional or leave the Action's argument a bit more flexible
0: owner's team (default)
>0: specific in-game house index
-1 all AI
-2 all non-allied AI
-3 all allied AI
...

**Comment by NetsuNegi** [comment]
> Why not iterate through all units and send the hunt signal? Probably the units should leave the AI Team's Taskforce before and then send the hunt mission. In that way you can check specific units and ignore non-combatants like harvesters. At least is what I thought initially.
> 
> If that would be a new script type action then it should affect by default only the owner's team and make "ALL AI players" optional or leave the Action's argument a bit more flexible 0: owner's team (default)
> 
> > 0: specific in-game house index. If I'm  not wrong index 0 is the human player that always is present in-game.
> > -1 all AI
> > -2 all non-allied AI
> > -3 all allied AI
> > ...

At first I tried the AllToHunt, but that's too uncontrollable. XD
I agree with you.

**Comment by FS-21** [comment]
I could try creating an action related to this when I have time.

Maybe this is interesting combined with conditional jumps >:-D

**Comment by Zorbung** [comment]
I hope you guys complete this, it's too fun to ignore it. It could also become essential in keeping AI base clean.
